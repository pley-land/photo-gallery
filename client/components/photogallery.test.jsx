import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import PhotoGallery from './photogallery';

Enzyme.configure({ adapter: new Adapter() });
describe('PhotoGallery Test Suite', () => {
  it('PhotoGallery renders correctly', () => {
    const tree = renderer
      .create(<PhotoGallery />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('PhotoGallery has 3 photo elements', () => {
    expect(mount(<PhotoGallery />).find('.photo')).toHaveLength(3);
  });

  it('PhotoGallery handles click', () => {
    const e = {
      type: 'click',
      timeStamp: 1532753006757,
      jQuery33103509413025485584: true,
    };
    const wrapper = shallow(<PhotoGallery />);
    expect(wrapper.instance().state.isOpen).toEqual(false);
    wrapper.instance().onClickHandler(e);
    expect(wrapper.instance().state.isOpen).toEqual(true);
  });

  it('check state', () => {
    const wrapper = shallow(<PhotoGallery />);
    expect(wrapper.find('#img1').exists).toBeTruthy();
    expect(wrapper.find('#img2').exists).toBeTruthy();
    expect(wrapper.find('#img3').exists).toBeTruthy();

    // Testing Initial State
    // console.log(wrapper.find('#img1'));
    //.simulate('onMouseEnter');
    // Testing state after mouseleave

    // Testing state after mouseover
  });
});
