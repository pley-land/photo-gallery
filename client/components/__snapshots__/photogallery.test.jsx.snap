// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGallery Test Suite PhotoGallery renders correctly 1`] = `
<div
  id="topshelf"
>
  <img
    alt="Searchbar"
    id="searchbar"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    id="top"
  >
    <img
      alt="mapIMG"
      id="mapIMG"
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <div
      id="photo-gallery"
    >
      <div
        id="photo1"
      >
        <img
          alt=""
          className="default photo"
          id="img1"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          src=""
        />
      </div>
      <div
        id="photo2"
      >
        <img
          alt=""
          className="selected photo"
          id="img2"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          src=""
        />
      </div>
      <div
        id="photo3"
      >
        <span
          className="centered shadow"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <img
            alt="squares"
            id="squares"
            src={
              Object {
                "process": [Function],
              }
            }
          />
          <br />
          See All 3
        </span>
        <img
          alt=""
          className="default photo"
          id="img3"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          src=""
        />
      </div>
    </div>
  </div>
</div>
`;
